pkg/controller/networkpolicy/networkpolicy_controller.go

```diff
@@ -1080,8 +1080,8 @@ func (n *NetworkPolicyController) syncAppliedToGroup(key string) error {
 	// Retrieve all Pods matching the podSelector.
 	pods, err = n.podLister.Pods(appliedToGroup.Selector.Namespace).List(selector)
 	for _, pod := range pods {
-		if pod.Status.PodIP == "" {
-			// No need to process Pod when IPAddress is unset.
+		if pod.Spec.NodeName == "" {
```

Comment:
If typically Controller always receives a Pod event without NodeName (and IP), maybe we should optimize to ignore the event in updatePod()?

I do not mean we need to make the change in this PR.

Reply:
makes sense, I would create an issue to track the optimization as there could be more cases: if nodeName is not set, no need to trigger appliedToGroup; if PodIP is not set, no need to trigger addressGroup.